### PR TITLE
Implement OpenExitingFileForReadWrite

### PR DESCRIPTION
**Pull request by dimitry-**
Test: run OrbitBaseTests

### CODE REVIEW

File: src/OrbitBase/File.cpp [hunk 1]
```diff
@@ -92,6 +92,15 @@ ErrorMessageOr<unique_fd> OpenFileForWriting(const std::filesystem::path& path)
   return OpenFile(path, kOpenFlags, kOpenMode);
 }
 
+ErrorMessageOr<unique_fd> OpenExistingFileForReadWrite(const std::filesystem::path& path) {
```
Comment (dpallotti) [inline src/OrbitBase/File.cpp:95]: I would say to move it after `OpenNewFileForReadWrite` to keep the same order as in the `.h`.

**Review by dpallotti — APPROVED** [review]

File: src/OrbitBase/FileTest.cpp [hunk 2]
```diff
@@ -83,6 +83,19 @@ TEST(File, OpenNewFileForReadWrite) {
   ASSERT_TRUE(fd_or_error.has_value()) << fd_or_error.error().message();
 }
 
+TEST(File, OpenEsitingFileForReadWrite) {
```
Comment (florian-kuebler) [inline src/OrbitBase/FileTest.cpp:86]:
```suggestion
TEST(File, OpenExistingFileForReadWrite) {
```

**Review by florian-kuebler — APPROVED** [review]
Besides the typo, lgtm! thanks!

File: src/OrbitBase/FileTest.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitBase/FileTest.cpp:86]: done.

File: src/OrbitBase/File.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/File.cpp:95]: done.

**Pull request merged**